client/driver/docker.go

```diff
@@ -408,6 +409,14 @@ func (d *DockerDriver) Start(ctx *ExecContext, task *structs.Task) (DriverHandle
 	if err := mapstructure.WeakDecode(task.Config, &driverConfig); err != nil {
 		return nil, err
 	}
+
+	if strings.Contains(driverConfig.ImageName, "https://") {
```

Comment:
This transformation has to occur before the image is pulled and inspected (around line 496) and therefore before the `createContainer` method is called (around line 543).